### PR TITLE
[compile] Show breakdown of graph break

### PR DESCRIPTION
**Pull request by delock**
This PR extends https://github.com/microsoft/DeepSpeed/pull/6570 by showing a breakdown of graph breaks.  So we can see how graph breaks are distributed among different reasons.  An example of graph break output can be seen from the following workflow run https://github.com/microsoft/DeepSpeed/actions/runs/11199157962

### CODE REVIEW

File: tests/torch_compile/test_compile.py [hunk 1]
```diff
@@ -14,22 +14,11 @@
 
 torch._dynamo.config.cache_size_limit = 100
 
-import collections
-
 
 def get_dynamo_stats():
     # TODO: consider deepcopy'ing the entire counters struct and
     # adding a helper to do subtraction on it
```
Comment (tohtana) [inline tests/torch_compile/test_compile.py:20]: Is this comment still valid?

**Review by tohtana — APPROVED** [review]
@delock Thank you @delock, this is a very useful feature! I left a comment, but it looks good to me.

File: tests/torch_compile/test_compile.py (hunk 1 above)
Comment (delock) [reply tests/torch_compile/test_compile.py:20]: Thanks for the comments, I think they should be removed.  Will update.

**Comment by delock** [comment]
@tohtana the unneeded comments had been removed.  Also make the workflow to count zero2, there is no graph breaks for zero2 in this simple case.

https://github.com/microsoft/DeepSpeed/actions/runs/11269655884?pr=6601

**Pull request merged**